luaui/Widgets/export_defs.lua

```diff
@@ -0,0 +1,60 @@
+function widget:GetInfo()
+	return {
+		name = "Unitdefs JSON Export",
+		desc = "Exports UnitDefs into JSON files.",
+		author = "Wereii",
+		date = "June 2023",
+		license = "GNU GPL, v2 or later, BA/BAR only",
+		layer = 99999,
+		enabled = false
+	}
+end
+
+local export_folder_path = "json_export/"
+
+local function TableToFile(tbl, filename)
+	local file, err = io.open(filename, "w")
+	if not file then
+		Spring.Echo("Error opening file: " .. err)
+		return
+	end
+	file:write(Json.encode(tbl))
+	file:close()
+end
+
+local function ExportDefs()
+	if not VFS.FileExists(export_folder_path) then
+		Spring.CreateDir(export_folder_path)
+	end
+
+	for id, unitDef in pairs(UnitDefs) do
+		Spring.Echo(string.format("Exporting unitdef: %s", unitDef.name))
+		local tbl = {}
+
+		tbl["id"] = unitDef.id
+		-- embed higher-level "computed" fields like translatedHumanName, translatedTooltip, etc.
+		for field_name, value in pairs(unitDef) do
+			if type(value) ~= "table" or type(value) ~= "function" or type(value) ~= "nil" then
+				tbl[field_name] = value
+			end
+		end
+
+		-- flatten the UnitDef metatable data into plain table
+		for k, v in unitDef:pairs() do
+			tbl[k] = v
+		end
+
+		TableToFile(tbl, string.format("json_export/%s.json", unitDef.name))
```

Comment:
You defined base folder path as a constant above, I would use it here

Reply:
Good catch